student/views.py

```diff
@@ -1,3 +1,25 @@
-# from django.shortcuts import render
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.shortcuts import render
+from .models import Student
+from django.views.generic.edit import UpdateView
 
-# Create your views here.
+
+def account_update_success(request):
+    is_student = Student.is_student(request.user.id)
+    return render(request, 'account_settings_update_success.html', {"is_student": is_student})
+
+
+class UpdateStudentSettings(UserPassesTestMixin, UpdateView):
+    model = Student
+    fields = ("full_name", "email", "date_of_birth", "phone_number", "educational_institution", "major", "about",
+              "graduation_date")
+    template_name = 'student_account_settings.html'
+    success_url = "/account_update_success"
+
+    def test_func(self):
+        return self.request.user.id == int(self.kwargs['pk'])
```

Comment:
what do you test here?

Reply:
@guy9050 
It's not a test (like a pytest test).
The abstract class 'UserPassesTestMixin' allows you to render a view only if the user that sent the request passes a certain test. the 'test_func' method is an abstract method of this class that has to be implemented when inheriting from it.
It checks here that the ID of the user that sent the request is equal to the ID in the URL that they're trying to get to.
This way you can only edit the account details of your own account, and you cannot access any other accounts by modifying the URL.

Reply:
thanks @guyy13